### ISSUE TITLE
Create rolling R squared quantiles function for visualizing predictive power

### ISSUE THREAD

**Issue by jeanmarcgp** [jeanmarcgp/xtsanalytics#20]
Function name TBD as the specifications come together.

The general idea is to understand the singular and joint predictive power of 1 or 2 features (jointly) against a target variable, by classifying the target variable into quantiles (2 or 3 classes typical, but possibly more)

The quantiles use a rolling window to make their classification, and the thresholds are specified with a vector argument  i.e. thresholds = 0.50 for 2 classes, or thresholds = c(0.25, 0.75) for 3 classes with a large middle one, etc.  This allows to explore unbalanced classes and arbitrary quantile sizes.

The X and Y correspond to each feature, and the quantile classes are associated with a dot color so it's easy to visualize.

In addition, the data can be split into timeframes of specific sizes (use a vector of dates), or into N equal sized windows.  The data may be plotted in a matrix for multiple timeframes to keep it simple.  Use ggplot2 to plot the data and add a linear regression line for each class.

Each classes linear regression line has an associated R squared, which can be reported in a matrix that is returned from the function call.

In other words, for each quantile class, we get an R squared by timeframes returned as a matrix.  This allows to automate the analysis over lots of different feature sets.

Last, this function should also work in 1 dimension?  Or maybe this is a different function?

This requires a clean specification because it's probably a good idea to architect this into a couple of different functions.
